frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx

```diff
@@ -134,31 +159,72 @@ const LogsTabForPodName: React.FC<{ podName: string }> = ({ podName }) => {
                 'Error loading logs'
               ) : !logsLoaded ? (
                 <>
-                  <Spinner size="sm" /> Loading log
+                  <Spinner size="sm" /> {isSmallScreen() ? 'Loading' : 'Loading log'}
                 </>
               ) : isPaused ? (
                 <>
-                  <PlayIcon /> Resume refreshing
+                  <PlayIcon /> {isSmallScreen() ? 'Resume' : 'Resume refreshing'}
                 </>
               ) : (
                 <>
-                  <PauseIcon /> Pause refreshing
+                  <PauseIcon /> {isSmallScreen() ? 'Pause' : 'Pause refreshing'}
                 </>
               )}
             </Button>
           </SplitItem>
           <SplitItem isFilled style={{ textAlign: 'right' }}>
-            <Tooltip position="top" content={<div>Download currently selected container</div>}>
-              <Button
-                onClick={onDownload}
-                variant="link"
-                aria-label="Download current logs"
-                icon={<DownloadIcon />}
-                isDisabled={!canDownload}
-              >
-                Download
-              </Button>
-            </Tooltip>
+            {downloading ? (
+              <>
+                <Spinner size="sm" />{' '}
+              </>
+            ) : (
+              ''
+            )}
+            {podContainers.length !== 0 ? (
+              <Dropdown
+                removeFindDomNode
+                toggle={
+                  isSmallScreen() ? (
+                    <KebabToggle
+                      onToggle={() => setIsDownloadDropdownOpen(!isDownloadDropdownOpen)}
+                    />
+                  ) : (
+                    <DropdownToggle
+                      id="download-steps-logs-toggle"
+                      onToggle={() => setIsDownloadDropdownOpen(!isDownloadDropdownOpen)}
+                    >
+                      Download
+                    </DropdownToggle>
+                  )
+                }
+                isOpen={isDownloadDropdownOpen}
+                isPlain={isSmallScreen()}
+                dropdownItems={[
+                  <DropdownItem key="current-container-logs" onClick={onDownload}>
+                    <Truncate
+                      content={isSmallScreen() ? 'Download current step log' : 'Current step log'}
+                    />
+                  </DropdownItem>,
+                  <DropdownItem key="all-container-logs" onClick={onDownloadAll}>
+                    <Truncate
+                      content={isSmallScreen() ? 'Download all step logs' : 'All step logs'}
```

Comment:
yes, according to the mocks here : https://github.com/opendatahub-io/odh-dashboard/pull/2034#discussion_r1383003323 since we don't have a "Download" dropdown but a kebab menu instead.